### DISCUSSION TITLE
where does my compiled firmware go?

### DISCUSSION THREAD

**Discussion by jphleba** [Jyers/Marlin#1430] [Q&A] [answered]
Hello,
I just successfully compiled the source using visual studio, platform IO, and Marlin auto build. I needed to add linear advance for my machine. I have not messed with visual studio much, and have no idea where to find the successfully compiled firmware. Where would it go by default? Thank you

**Comment by ashefpv** [comment, marked as answer]
It will go to the ".pio" directory of the first Workspace folder that you have open. In my case that is:

D:\storage\3D Printer\Firmware\JyersUI-Marlin-Ender_3_V2_202109262042\.pio\build\STM32F103RET6_creality\firmware-20210926-220612.bin

![image](https://user-images.githubusercontent.com/47304999/134856537-a5ac2bce-147a-4312-8c96-26b574b8f76a.png)

When in doubt, just search for "*.bin" files and you'll find it.

**Reply by gilegue** [reply]
If you use Marlin Auto Build, it should be showing near your preferred environment (under your chosen environment to be exact). Just click on those and a new explorer will pop up exactly in your .bin folder. 
![image](https://user-images.githubusercontent.com/6836710/135503747-ec274c57-4eb9-484f-96d2-5f607cb73194.png)
Just click on those red box and you're done
